ElliotForWater/elliotforwater.com
# Add recaptcha

### Related Issue
resolves #47

## Description
Added Google Recaptcha v2 to subscribe forms.

With a valid Recaptcha site key, verification will show:
![image](https://user-images.githubusercontent.com/47090434/146310955-cffa8f4b-0eb7-4bc9-b100-e7fd914c1f7f.png)

Without Recaptcha verification, form will not submit and an error toast will show:
![image](https://user-images.githubusercontent.com/47090434/146311082-f24acb55-f6be-4d59-8893-4ac4a50721c9.png)

When form is submitted successfully, the recaptcha token is also sent with the form data:
![image](https://user-images.githubusercontent.com/47090434/146311312-34e6b817-9fe5-44fd-8f89-31c3bb481f81.png)

Docs for verifying token on the server: https://developers.google.com/recaptcha/docs/verify


#### Impacted Areas in Application
Subscribe Forms


#### Steps to Test or Reproduce
Sign into google and get a site key. Make sure to add 'localhost' as a domain: https://www.google.com/recaptcha/admin/create

Replace 'FAKE KEY' with your site key:
![image](https://user-images.githubusercontent.com/47090434/146311881-3a565e35-8ef0-4685-9d8f-d16b28bf948d.png)

#### Related PRs
N/A

------------------------------------------------
## PR Checklist:
